code/datums/elements/footstep.dm

```diff
@@ -92,6 +94,9 @@
 /datum/element/footstep/proc/play_simplestep(mob/living/source)
 	SIGNAL_HANDLER
 
+	if (SHOULD_DISABLE_FOOTSTEPS(source))
```

Comment:
isn't this kind of lazy in that it still requires a proc call?

really teh element should get beleted and the signals deregged

Reply:
I asked MSO about this as well, here's my collected thoughts:

The big performance loss was not in the fact that this is being called every Moved, but the playing of sounds specifically.

We want to be able to easily toggle footsteps on/off, but we also want people with TRAIT_EXEMPT_WHATEVER_IT_WAS to still have footsteps.

This is the easiest way to do all that cleanly, and solves the big performance loss.

Reply:
okay i trust you
my child would type knife